freqtrade/resolvers/strategy_resolver.py

```diff
@@ -123,7 +121,8 @@ def _load_strategy(
         current_path = Path(__file__).parent.parent.joinpath('strategy').resolve()
 
         abs_paths = [
-            config['user_data_dir'].joinpath('strategies'),
+            Path(config['user_data_dir']).joinpath('strategies'),
+            config['strategy_path'],
```

Comment:
As mentioned in the other PR - adding  'strategy_path' here will result in searching this branch twice, since it's already passed in as `extra_dir`.

Reply:
Fixed. I deleted the extra strategy path but left in the Path wrapper because:
```python
print(config['user_data_dir'].joinpath('strategies'))

AttributeError: 'str' object has no attribute 'joinpath'
```

Reply:
in this case, your setting of `config['user_data_dir']` is wrong / you're probably using it from a notebook, which may cause problems since not everything is initialized correctly.

The following lines assign user_data_dir with the result of `create_userdata_dir()` ... which always returns a Path object.

https://github.com/freqtrade/freqtrade/blob/f987e6e0f9bdcae6f0cec3a273e270a3a35a3c12/freqtrade/configuration/configuration.py#L169-L176


Using config within a notebook will still need some more love / work (it's on my long todo-list) - but i would not cover the problems caused by this in other areas of the code if they cannot happen (all regular operations do run through all / most of the sections in the configuration class).

Reply:
That is the problem. In the notebook, I was using:
```python
candles = load_pair_history(datadir=config['datadir'],
                            ticker_interval=ticker_interval,
                            pair=pair)
```
The dict index retrieves the datadir as a str. I can work around it by wrapping the argument with `Path()`. I will be interested to see how you want to handle calling internal functions from a notebook. It would be nice to arbitrarily pass the Configuration object to any function.